homeassistant/components/tuya/config_flow.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+"""Config flow for Tuya."""
+
+import logging
+
+from tuya_iot import ProjectType, TuyaOpenAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import (
+    CONF_ACCESS_ID,
+    CONF_ACCESS_SECRET,
+    CONF_APP_TYPE,
+    CONF_COUNTRY_CODE,
+    CONF_ENDPOINT,
+    CONF_PASSWORD,
+    CONF_PROJECT_TYPE,
+    CONF_USERNAME,
+    DOMAIN,
+    TUYA_APP_TYPE,
+    TUYA_ENDPOINT,
+    TUYA_PROJECT_TYPE,
+)
+
+RESULT_SINGLE_INSTANCE = "single_instance_allowed"
```

Comment:
Why is only a single instance allowed? What if someone has two homes?

Reply:
All homes devices that belong to one account will be fetched from tuya-iot-python-sdk after login.

Reply:
Sure. But that still doesn't mean we can't have 2 accounts configured ?

Reply:
Yes, multi-accounts are not supported now. But we will consider it in the future.